ext/semian/semian.c

```diff
@@ -202,14 +142,14 @@ configure_tickets(int sem_id, int tickets, int should_initialize)
   }
 }
 
-static int
-create_semaphore(int key, int permissions, int *created)
+int
+create_semaphore(int key, long permissions, int *created)
 {
   int semid = 0;
   int flags = 0;
 
   *created = 0;
-  flags = IPC_EXCL | IPC_CREAT | FIX2LONG(permissions);
```

Comment:
This check was happening in the wrong spot.

We were truncating long -> int and then going back to long. I've fixed this in the correct spot.